ducktape/tests/reporter.py

```diff
@@ -18,45 +18,109 @@
 import pkg_resources
 
 
-class TestReporter(object):
+SEPARATOR_LENGTH = 100
+
+
+def pass_fail(success):
+    """Convenient helper. Converts boolean to PASS/FAIL."""
+    return "PASS" if success else "FAIL"
+
+
+def format_time(t):
+    """Return human-readable interval of time.
+    Assumes t is in units of seconds.
+    """
+    minutes = int(t / 60)
+    seconds = t % 60
+
+    r = ""
+    if minutes > 0:
+        r += "%d minute%s " % (minutes, "" if minutes == 1 else "s")
+    r += "%.3f seconds" % seconds
+    return r
+
+
+class SingleTestReporter(object):
+    """Helper class for creating a view of results from a single test."""
+
+    def __init__(self, result):
+        self.result = result
+
+    def header_string(self):
+        """Header lines of the report"""
+        header_lines = [
+            "=" * SEPARATOR_LENGTH,
+            "test_id:    %s" % self.result.test_context.test_id,
+            "run time:   %s" % format_time(self.result.run_time),
+            "status:     %s" % pass_fail(self.result.success),
+            "=" * SEPARATOR_LENGTH
+        ]
+
+        return "\n".join(header_lines)
+
+    def result_string(self):
+        """Stringify a single result."""
+        result = self.result
+        result_lines = []
+
+        if result.data is not None:
+            result_lines.append(json.dumps(result.data))
```

Comment:
We should probably be catching failures to serialize and doing something useful with them (unless they get caught ok here and the normal exception cleanup code handles this case?). Would be nice to warn the user why they're getting this in case they include something non-JSON-serializable in their output.